tests/processes/ProcessingTest.py

```diff
@@ -226,6 +227,77 @@ def test_process_queues(self):
         self.assertEqual(self.queue.get(timeout=0), ([first_global]))
         self.assertEqual(self.queue.get(timeout=0), ([first_global]))
 
+    def test_loaded_bears_with_error(self):
+        count_expected_bears = 3
+        count_loaded_bears = 2
+        all_files = [self.testcode_c_path]
+        cache = FileCache(self.log_printer, 'coala_test', flush_cache=True)
+        cache.track_files(all_files)
+        uncache_if_unloaded_bears(count_expected_bears,
+                                  count_loaded_bears,
+                                  all_files,
+                                  cache,
+                                  self.log_printer)
+
+        cache.write()
+        self.assertFalse(self.testcode_c_path in cache.data)
+
+        count_expected_bears = 3
+        count_loaded_bears = 3
+        all_files = [self.testcode_c_path]
+        cache = FileCache(self.log_printer, 'coala_test', flush_cache=True)
+        cache.track_files(all_files)
+        uncache_if_unloaded_bears(count_expected_bears,
+                                  count_loaded_bears,
+                                  all_files,
+                                  cache,
+                                  self.log_printer)
+        cache.write()
+        self.assertTrue(self.testcode_c_path in cache.data)
+
+    def test_loaded_bears_with_error_result(self):
+        class BearWithMissingPrerequisites(Bear):
+
+            def __init__(self, section, queue, timeout=0.1):
+                Bear.__init__(self, section, queue, timeout)
+
+            def run(self):
+                return []
+
+            @classmethod
+            def check_prerequisites(cls):
+                return False
+
+        multiprocessing.Queue()
+        tmp_local_bears = copy.deepcopy(self.local_bears['cli'])
+        tmp_local_bears.append(BearWithMissingPrerequisites)
+        cache = FileCache(self.log_printer,
+                          'coala_test_on_error',
+                          flush_cache=True)
+        results = execute_section(self.sections['cli'],
+                                  [],
+                                  tmp_local_bears,
+                                  lambda *args: self.result_queue.put(args[2]),
+                                  cache,
+                                  self.log_printer,
+                                  console_printer=self.console_printer)
+        cache.write()
+        self.assertLess(len(cache.data),
+                        len(self.sections['cli'].get('files', '')))
```

Comment:
please add a newline here, after this stmt you make a new setup for another assertion^^